### PR TITLE
feat: Simple dev tools plugin with support for opening H2 console

### PR DESCRIPTION
**Pull request by Artur-**

### CODE REVIEW

**Comment by Artur-** [comment]
This should be merged after #1485 as it depends on the project setup for `hilla-dev-mode`

File: packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java [hunk 1]
```diff
@@ -70,5 +78,27 @@ public void execute() throws ExecutionFailedException {
...
+                        }
+                        """;
+                FileUtils.writeStringToFile(vaadinTs, current,
+                        StandardCharsets.UTF_8);
+            }
+        } catch (IOException e) {
+            throw new ExecutionFailedException("Failed to update vaadin.ts", e);
+
+        }
+
```
Comment (taefi) [inline packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java:102]: Couldn't an `if (productionMode) { return; }` prevent opening and reading `vaadinTs` when it is building for production?

File: packages/java/hilla-dev-mode/src/main/resources/META-INF/resources/frontend/dev-tools-database.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    const h2Window = window.open(this.h2path)!;
+    await waitForConsole(h2Window);
+    loginUsingUrl(h2Window, this.h2jdbcUrl);
+  }
+}
+
+const plugin: DevToolsPlugin = {
+  init(devToolsInterface: DevToolsInterface): void {
+    devTools = devToolsInterface;
+    devTools.addTab('Database', 'devtools-database');
```
Comment (taefi) [inline packages/java/hilla-dev-mode/src/main/resources/META-INF/resources/frontend/dev-tools-database.ts:63]:
```suggestion
    devTools.addTab('H2 Database', 'devtools-database');
```

**Review by taefi — CHANGES_REQUESTED** [review]
It seems that there are two unrelated features being added in a single PR, right? 
1. Hilla HotSwap Plugin
2. A dev tools plugin for opening H2 console

File: packages/java/hilla-dev-mode/src/main/resources/META-INF/resources/frontend/dev-tools-database.ts (hunk 2 above)
Comment (Artur-) [reply packages/java/hilla-dev-mode/src/main/resources/META-INF/resources/frontend/dev-tools-database.ts:63]: I would keep it as "database" as the intent is to add more functionality there, like example data generation. This is just a first step

File: packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java (hunk 1 above)
Comment (Artur-) [reply packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java:110]: Is that better?

Comment (Artur-) [reply packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java:110]: The current version does not read vaadinTs in production mode either

**Review by taefi — APPROVED** [review]

File: packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java (hunk 1 above)
Comment (Artur-) [reply packages/java/engine-runtime/src/main/java/dev/hilla/internal/TaskGenerateEndpointImpl.java]: Extracted to a method to make it cleaner

**Review by taefi — APPROVED** [review]

**Pull request merged**